docs/configuration/pipeline-configuration-files.mdx

```diff
@@ -0,0 +1,96 @@
+    status: running             # pipelines status at startup, either running or stopped. [mandatory]
+    name: pipeline1             # pipeline name, if not specified, pipeline ID will be used as name. [optional]
+    description: desc           # pipeline description. [optional]
+    connectors:                 # a map of connectors IDs and their configurations.
+      con1:                     # connector ID, has to be unique per pipeline.
+        type: source            # connector type, either "source" or "destination". [mandatory]
+        plugin: builtin:file    # connector plugin. [mandatory]
+        name: con3              # connector name, if not specified, connector ID will be used as name. [optional]
+        settings:               # map of configurations keys and their values.
+          path: ./file1.txt     # for this example, the plugin "bultin:file" has only one configuration, which is path.
+      con2:
+        type: destination
+        plugin: builtin:file
+        name: file-dest
+        settings:
+          path: ./file2.txt
+        processors:             # a map of processor IDs and their configurations, "con2" is the processor parent.
+          proc1:                # processor ID, has to be unique for each parent
+            type: js            # processor type. [mandatory]
+            settings:           # map of processor configurations and values
+              Prop1: string
+    processors:                 # processor IDs, that have the pipeline "pipeline1" as a parent.
+      proc2:
+        type: js
+        settings:
+          prop1: ${ENV_VAR}     # yon can use environmental variables by wrapping them in a dollar sign and curly braces ${}.
+```
+
+If the file is invalid (missed a mandatory field, or has an invalid configuration value), then the pipeline that has the
+invalid value will be skipped, with an error message logged.
+
+If two pipelines in one file have the same ID, or the `version` field was not specified, then the file would be
+non-parsable and will be skipped with an error message logged.
+
+If two pipelines from different files have the same ID, the second pipeline will be skipped, with an error message
+specifying which pipeline was not provisioned.
+
+**_Note_**: Connector IDs and processor IDs will get their parent ID prefixed, so if you specify a connector ID as `con1`
+and its parent is `pipeline1`, then the provisioned connector will have the ID `pipeline1:con1`. Same goes for processors,
+if the processor has a pipeline parent, then the processor ID will be `connectorID:processorID`, and if a processor
+has a connector parent, then the processor ID will be `pipelineID:connectorID:processorID`.
+
+## Pipelines Immutability
+Pipelines provisioned by configuration files are **immutable**, any updates needed on a provisioned pipeline have to be
+done through the configuration file it was provisioned from. You can only control stopping and starting a pipeline
```

Comment:
Nitpick: "any updates needed on a provisioned pipeline have to be done **through the configuration file it was provisioned from**" At least to me, this indicates that the changes need to be done through the same file the pipeline was provisioned from. We should probably say something along the lines of "any updates need to be done through a configuration file", to state that all it matters is that the pipeline is changed through a file, and not through the API.